src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -269,6 +287,37 @@ internal static int NumberOfValueTuples(int numElements, out int remainder)
         /// <summary>
         /// Produces the underlying ValueTuple corresponding to this list of element types.
         /// </summary>
+        private static NamedTypeSymbol GetTupleUnderlyingType(ImmutableArray<TypeSymbol> elementTypes, CSharpCompilation compilation)
```

Comment:
It feels like there is a lot of code is getting duplicated here. The function below can be changed to 
- take a compilation instead of binder, logic in Binder.GetWellKnownType isn't that complex to duplicate, just a Binder.ReportUseSiteDiagnostics call after Compilation.GetWellKnownType;
- report diagnostics only if bag is not null.